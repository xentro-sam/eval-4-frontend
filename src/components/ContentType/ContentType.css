#content-type {
    color: rgb(130, 130, 130);
    font-size: 1vw;
    font-weight: 600;
    border-radius: 5px;
    padding: 1vw;
    margin: 0.5vw;
    background-color: white;
    display: flex;
    justify-content: space-between;
}