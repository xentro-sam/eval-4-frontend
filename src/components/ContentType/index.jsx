import * as React from 'react';
import './ContentType.css';
import PropTypes from 'prop-types';
import makeRequest from '../../utils/makeRequest';
import {GET_CONTENT_TYPE_FIELDS as getContentTypeFields} from '../../constants/apiEndPoints';

export default function ContentType(props) {
  const [fieldCount, setFieldCount] = React.useState(0);
  const [fields, setFields] = React.useState([]);
  React.useEffect(() => {
    makeRequest(getContentTypeFields(props.id), {})
        .then((response) => {
          setFields(response);
          setFieldCount(response.length);
        });
  }, []);
  const handleClick = () => {
    props.setFields(fields);
    props.setContainerTitle(props.contentTypeName);
    props.setContentTypeId(props.id);
  };

  return (
    <div id="content-type" onClick={handleClick}>
      <div id="content-type-name">{props.contentTypeName}</div>
      <div id="content-type-count">{fieldCount}</div>

    </div>
  );
}

ContentType.propTypes = {
  contentTypeName: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  setFields: PropTypes.func.isRequired,
  setContainerTitle: PropTypes.func.isRequired,
  setContentTypeId: PropTypes.func.isRequired,
};
