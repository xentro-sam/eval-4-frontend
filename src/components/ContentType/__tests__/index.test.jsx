import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContentType from '../index';
import {waitFor} from '@testing-library/dom';
import makeRequest from '../../../utils/makeRequest';
import {act} from 'react-dom/test-utils';

jest.mock('../../../utils/makeRequest');

const mockData = {
  contentTypeName: 'mock title 1',
  id: 1,
  setFields: jest.fn(),
  setContainerTitle: jest.fn(),
  setContentTypeId: jest.fn(),
};

const mockResponse = [
  'mock title 1',
  'mock title 2',
  'mock title 3',
];

describe('ContentType', () => {
  it('should render correctly', async () => {
    makeRequest.mockResolvedValueOnce(mockResponse);
    expect(makeRequest).not.toHaveBeenCalled();
    const {asFragment} = await act( async () => render(<ContentType {...mockData} />));
    await waitFor(() => {
      expect(screen.getByText('mock title 1')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
    expect(makeRequest).toBeCalledTimes(1);
  });
});
