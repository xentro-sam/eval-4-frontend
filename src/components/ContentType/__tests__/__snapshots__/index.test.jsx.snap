// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    id="content-type"
  >
    <div
      id="content-type-name"
    >
      mock title 1
    </div>
    <div
      id="content-type-count"
    >
      3
    </div>
  </div>
</DocumentFragment>
`;
